sqs/src/main/scala/akka/stream/alpakka/sqs/SqsSourceStage.scala

```diff
@@ -19,20 +19,69 @@ object SqsSourceSettings {
   def create(waitTimeSeconds: Int, maxBufferSize: Int, maxBatchSize: Int): SqsSourceSettings =
     SqsSourceSettings(waitTimeSeconds, maxBufferSize, maxBatchSize)
 
+  def create(waitTimeSeconds: Int,
+             maxBufferSize: Int,
+             maxBatchSize: Int,
+             attributeNames: util.List[String],
+             messageAttributeNames: util.List[String]): SqsSourceSettings =
+    SqsSourceSettings(waitTimeSeconds,
+                      maxBufferSize,
+                      maxBatchSize,
+                      attributeNames.asScala,
+                      messageAttributeNames.asScala)
+
 }
 
 //#SqsSourceSettings
 final case class SqsSourceSettings(
     waitTimeSeconds: Int,
     maxBufferSize: Int,
-    maxBatchSize: Int
+    maxBatchSize: Int,
+    attributeNames: Seq[String] = Seq(),
+    messageAttributeNames: Seq[String] = Seq()
 ) {
   require(maxBatchSize <= maxBufferSize, "maxBatchSize must be lower or equal than maxBufferSize")
   // SQS requirements
   require(0 <= waitTimeSeconds && waitTimeSeconds <= 20,
           s"Invalid value ($waitTimeSeconds) for waitTimeSeconds. Requirement: 0 <= waitTimeSeconds <= 20 ")
   require(1 <= maxBatchSize && maxBatchSize <= 10,
           s"Invalid value ($maxBatchSize) for maxBatchSize. Requirement: 1 <= maxBatchSize <= 10 ")
+
+  private val validNames = Seq(
```

Comment:
How about enumerating them as an ADT instead of stringly typed: 
```
sealed abstract class AttributeName(val name: String)
case object All extends AttributeName("All")
```

and then `attributeNames: Set[AttributeName]` etc. ?